packages/good-design/src/hooks/useFormatTransactions.ts

```diff
@@ -0,0 +1,68 @@
+import { useMemo } from "react";
+import { G$Amount, PoolDetails, SupportedChains, RecentClaims, useGetEnvChainId } from "@gooddollar/web3sdk-v2";
+import { BigNumber } from "ethers";
+
+import { truncateMiddle } from "../utils";
+import type { ClaimContextProps } from "../apps/ubi/types";
+
+export const useFormatClaimTransactions = (
+  pools: PoolDetails[] | RecentClaims[] | undefined,
+  chainId: number | undefined,
+  account: string
+): ClaimContextProps["claimPools"] => {
+  const { defaultEnv } = useGetEnvChainId();
+
+  return useMemo(() => {
+    if (!pools || pools.length === 0 || !chainId) return undefined;
+
+    const formattedTransactions: any = { totalAmount: BigNumber.from("0"), transactionList: [] };
+
+    pools?.map((pool: PoolDetails | RecentClaims) => {
+      const [poolName] = Object.keys(pool);
+      const poolDetail = Object.values(pool)[0];
```

Comment:
I dont understand this.
Also why is poolDetail is then processed as transactions?

Reply:
this formatter hook is used both for pre and post-claim lists/pools and displaying them in a list of `TransactionCards`.

pooldetails here contains pre-claim data, or contains post-claim transaction list.

maybe it could use a different naming for clarity?